packages/iov-ethereum/src/ethereumconnection.ts

```diff
@@ -712,7 +713,9 @@ export class EthereumConnection implements AtomicSwapConnection {
     }
   }
 
-  public listenTx(query: TransactionQuery): Stream<ConfirmedTransaction | FailedTransaction> {
+  public listenTx(
+    query: TransactionQuery,
+  ): Stream<ConfirmedTransaction<LightTransaction> | FailedTransaction> {
```

Comment:
Great! When you now change the return type of `searchSendTransactionsByAddressOnScraper`, we get rid of this extra network query for each transaction: `const transaction = (await this.searchTransactionsById(Parse.transactionId(tx.hash)))[0];`